include/OsmAndCore/Utilities.h

```diff
@@ -889,6 +889,32 @@ namespace OsmAnd
         static LatLon rhumbDestinationPoint(LatLon latLon, double distance, double bearing);
         static std::pair<int, int> calculateFinalXYFromBaseAndPrecisionXY(int bazeZoom, int finalZoom, int precisionXY,
                                                                           int xBase, int yBase, bool ignoreNotEnoughPrecision);
+
+        inline static void resizeVector(const PointF& start, PointF& end, float sizeIncrement)
+        {
+            auto vec = end - start;
+            auto vecLength = vec.norm();
+            auto vecDir = vec/vecLength;
+            end = start + vecDir * (vecLength + sizeIncrement);
+        }
+
+        inline static PointF computeNormalToLine(const PointF& start, const PointF& end, int8_t dir)
```

Comment:
It could've been template for Point<T>, there's no reasoning for dir to be int8_t.

On top of that, you can just select one and callee can invert `-computeNormalToLine()` if needed.

Reply:
> It could've been template for Point

Please find common ground with @vshcherb , regarding to usage of templates.
It was implemented as a template few iteration before.
https://github.com/osmandapp/OsmAnd-core/pull/261/commits/2fe690f9aee67feff8ba4490024f7dfd07b5e4b2#r747365804


> there's no reasoning for dir to be int8_t.

What is the reason for usage of other type instead of `int8_t`?

Reply:
@vshcherb what's your argument not to use Point<T>?

In this method, direction is binary value, and as I've said - I see no reason to have the if block in the method at all